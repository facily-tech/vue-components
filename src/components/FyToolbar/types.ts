export interface IItemsToolbar {
  icon?: string;
  title?: string;
  options?: {
    label: string;
  }[];
}
