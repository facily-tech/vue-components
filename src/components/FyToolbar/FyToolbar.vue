<template>
  <v-toolbar color="transparent" v-bind="propsAndAttrs" :flat="flat" class="mb-4">
    <v-toolbar-title class="headline font-weight-medium" v-if="screenWidth > 768">
      {{ items.title }}
    </v-toolbar-title>
    <v-spacer></v-spacer>
    <fy-button v-if="backButton" @click="back" class="pa-3" outlined color="primary">
      <v-icon> mdi-arrow-left </v-icon>
      Voltar
    </fy-button>
    <div class="ml-2">
      <slot name="btn"></slot>
    </div>
    <v-menu offset-y v-if="items.options">
      <template v-slot:activator="{ on, attrs }">
        <fy-button icon color="primary" v-bind="attrs" v-on="on">
          <v-icon>mdi-dots-vertical</v-icon>
        </fy-button>
      </template>
      <v-list>
        <v-list-item v-for="(option, index) in items.options" :key="index">
          <v-list-item-title>
            {{ option.label }}
          </v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
  </v-toolbar>
</template>

<script lang="ts">
import Vue from 'vue';
import { IItemsToolbar } from './types';
import FyButton from '@/components/FyButton/FyButton.vue';

export default Vue.extend({
  name: 'fy-toolbar',

  components: { FyButton },

  props: {
    items: {
      type: Object as () => IItemsToolbar,
      default: {} as IItemsToolbar,
      required: true,
    },
    backButton: {
      type: Boolean,
      default: false,
    },
    flat: {
      type: Boolean,
      default: true,
    },
  },

  data() {
    return {
      screenWidth: 0,
    };
  },

  created(): void {
    this.changeScreen();
    window.addEventListener('resize', () => {
      this.changeScreen();
    });
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },

  methods: {
    back(): void {
      this.$emit('back-event');
    },
    changeScreen(): void {
      this.screenWidth = window.screen.width;
    },
  },
});
</script>
