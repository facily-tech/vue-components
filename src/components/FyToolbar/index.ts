export { default as FyToolbar } from './FyToolbar.vue';
