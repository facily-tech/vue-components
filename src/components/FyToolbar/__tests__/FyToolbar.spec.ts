import Vue from 'vue';
import Vuetify from 'vuetify';
import FyToolbar from '../FyToolbar.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyToolbar.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyToolbar>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyToolbar, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
