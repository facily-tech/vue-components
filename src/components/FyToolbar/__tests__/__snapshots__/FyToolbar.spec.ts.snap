// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyToolbar.ts should render component and match snapshot 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--flat" style="height: 64px;" items="[object Object]">
  <div class="v-toolbar__content" style="height: 64px;"><i aria-hidden="true" class="v-icon notranslate mr-2 material-icons theme--light" style="display: none;"></i>
    <!---->
    <div class="spacer"></div>
    <!---->
    <div class="ml-2"></div>
    <!---->
  </div>
</header>
`;
